src/messenger.rs

```diff
@@ -469,6 +470,14 @@ where
 
         Err(SyncVersionsError::NoWorkingVersion)
     }
+
+    pub async fn sasl_handshake(&self, sasl_config: &SaslConfig) -> Result<(),  RequestError> {
+        let req = SaslHandshakeRequest::new();
+        self.request(req).await?;
+        let req = SaslAuthenticateRequest::new(sasl_config);
+        self.request(req).await?;
+        Ok(())
+    }
```

Comment:
So if there is a `SaslConfig`, then we need to call this each time after `sync_versions`. I think you can just grep throw the code and see where is this done. Maybe you wanna make the two methods private and instead offer a `setup` method that does both, so we don't forget to call one or the other a certain callsides.